### DISCUSSION TITLE
Trying to configure eslint/tseslint to lint only Playwright tests

### DISCUSSION THREAD

**Discussion by cviebrock** [playwright-community/eslint-plugin-playwright#328] [Q&A]
I have a project with lots of legacy JS code, that I don't really care about.  I do have a `e2e` folder in the root of the project with Playwright tests that I do want to lint, but I can't figure out how to set up my eslint config correctly.

I basically want it to only lint the (*.ts) files in the `e2e` directory and ignore everything else.

I tried the recommended config from this project, and that didn't work.  So where I am now is with a config based on what [typescript-eslint recommends](https://typescript-eslint.io/packages/typescript-eslint/#advanced-usage) when using other plugins, and [this answer on Stackoverflow](https://stackoverflow.com/questions/73915236/how-do-i-configure-eslint-to-ignore-my-root-folder-and-to-only-include-my-src-fo) about only processing one directory.

However, no matter what, running `./node_modules/.bin/eslint` seems to either lint *every* file, or complains that everything is ignored and there is nothing to process. 

Any suggestions for an eslint noob?

```js
// eslint.config.mjs

import eslint from '@eslint/js';
import tseslint from 'typescript-eslint';
import playwrightPlugin from 'eslint-plugin-playwright';

export default tseslint.config(
  {
    ignores: ['/*', '!/e2e']
  },
  eslint.configs.recommended,
  {
    plugins: {
      '@typescript-eslint': tseslint.plugin,
      playwright: playwrightPlugin
    },
    languageOptions: {
      parser: tseslint.parser,
      parserOptions: {
        projectService: true
      }
    },
    rules: {
      '@typescript-eslint/no-unsafe-argument': 'error',
      '@typescript-eslint/no-unsafe-assignment': 'error',
      '@typescript-eslint/no-unsafe-call': 'error',
      '@typescript-eslint/no-unsafe-member-access': 'error',
      '@typescript-eslint/no-unsafe-return': 'error'
    }
  },
  {
    files: ['e2e/**'],
    ...playwrightPlugin.configs['flat/recommended']
  }
);
```

Relevant `devDependencies` from `package.json`:
```json
{
    "@eslint/js": "^9.13.0",
    "@playwright/test": "^1.42.0",
    "@types/node": "^20.11.21",
    "eslint": "^9.13.0",
    "eslint-plugin-playwright": "^2.0.0",
    "globals": "^15.11.0",
    "typescript": "^4.7.4",
    "typescript-eslint": "^8.11.0"
}
```

**Comment by mskelton** [comment]
Why not just run `eslint e2e` as your script? Or just set the `src` or other folders as `ignores`?

**Reply by cviebrock** [reply]
I hadn't thought of that ... makes perfect sense.

But let's say I end up adding another folder I want to lint.  I could just do `eslint {e2e,newstuff}` I suppose ... but isn't there a way to handle this in the config?  Heck, if there were an `includes` option, alongside the `ignores`, that would be awesome. :)

**Reply by mskelton** [reply]
I'd recommend opening a issue in the main ESlint repo since this is not relevant specifically to the playwright plugin.  Your use case is definitely not common.
